### PR TITLE
support sd3.5 for controlnet example

### PR DESCRIPTION
**Pull request by DavyMorgan**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by DavyMorgan** [comment]
@sayakpaul Hi, in this PR, I add sd3.5 to the controlnet example, with only a few lines added!

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Comment by yiyixuxu** [comment]
@DavyMorgan can you fix the test?

**Comment by DavyMorgan** [comment]
> @DavyMorgan can you fix the test?

Yeah, I have updated the test.

**Comment by yiyixuxu** [comment]
@DavyMorgan 
would you be able to update the branch to sync with the main (and resolve the conflicts)?
sorry about that!

**Comment by DavyMorgan** [comment]
> @DavyMorgan would you be able to update the branch to sync with the main (and resolve the conflicts)? sorry about that!

Sure, I can fix that.

**Comment by DavyMorgan** [comment]
@yiyixuxu @sayakpaul Could you please run the test workflows again?

**Comment by DavyMorgan** [comment]
@sayakpaul I updated the codes. This time it should be ok. :)

File: examples/controlnet/train_controlnet_sd3.py [hunk 1]
```diff
@@ -986,7 +989,7 @@ def main(args):
         controlnet = SD3ControlNetModel.from_pretrained(args.controlnet_model_name_or_path)
     else:
         logger.info("Initializing controlnet weights from transformer")
-        controlnet = SD3ControlNetModel.from_transformer(transformer)
+        controlnet = SD3ControlNetModel.from_transformer(transformer, num_extra_conditioning_channels=0)
```
[the remark this answers is not in the document]
Comment (DavyMorgan) [reply examples/controlnet/train_controlnet_sd3.py]: Thanks, I have added a CLI arg.

**Comment by DavyMorgan** [comment]
> https://huggingface.co/hf-internal-testing/tiny-sd35-pipe

@sayakpaul I have updated the codes to use hf-internal-testing/tiny-sd35-pipe. Meanwhile, the example results can be found at the end of examples/controlnet/README_sd3.md.

File: src/diffusers/models/transformers/transformer_sd3.py [hunk 2]
```diff
@@ -344,7 +343,7 @@ def custom_forward(*inputs):
 
             # controlnet residual
             if block_controlnet_hidden_states is not None and block.context_pre_only is False:
-                interval_control = len(self.transformer_blocks) // len(block_controlnet_hidden_states)
+                interval_control = int(math.ceil(len(self.transformer_blocks) / len(block_controlnet_hidden_states)))
```
Comment (yiyixuxu) [inline src/diffusers/models/transformers/transformer_sd3.py:346]: sorry I missed this change - any reason we make this change here?

**Comment by yiyixuxu** [comment]
related ( a little bit overlap with) https://github.com/huggingface/diffusers/pull/9758

**Comment by DavyMorgan** [comment]
@yiyixuxu @sayakpaul Can you run the tests again to check whether it works well with the current HEAD?

**Comment by sayakpaul** [comment]
@DavyMorgan sorry for the late reply. Could you consider the changes introduced in https://github.com/huggingface/diffusers/pull/9758 and repurpose this PR accordingly?

File: src/diffusers/models/controlnets/controlnet_sd3.py [hunk 3]
```diff
@@ -56,6 +56,10 @@ def __init__(
         pooled_projection_dim: int = 2048,
         out_channels: int = 16,
         pos_embed_max_size: int = 96,
+        dual_attention_layers: Tuple[
```
Comment (yiyixuxu) [inline src/diffusers/models/controlnets/controlnet_sd3.py:59]: we already have this change https://github.com/huggingface/diffusers/blob/963ffca43419a8dffa682d9e03c2299b76feeced/src/diffusers/models/controlnets/controlnet_sd3.py#L60

can you rebase?

Comment (DavyMorgan) [reply src/diffusers/models/controlnets/controlnet_sd3.py]: @yiyixuxu done.

**Comment by DavyMorgan** [comment]
@yiyixuxu @sayakpaul Hi, I have fixed the codes to pass the quality test. Please run the tests again!

**Comment by DavyMorgan** [comment]
Sorry for missing the typo😂Now fixed.

**Pull request merged**